services/datamanager/builtin/builtin.go

```diff
@@ -343,95 +340,60 @@ func (svc *builtIn) getCollectorFromConfig(attributes dataCaptureConfig) (data.C
 	return nil, nil
 }
 
-func (svc *builtIn) initOrUpdateSyncer(_ context.Context, intervalMins float64, cfg *config.Config) error {
-	// If user updates sync config while a sync is occurring, the running sync will be cancelled.
-	// TODO DATA-235: fix that
-	if svc.syncer != nil {
-		// If previously we were syncing, close the old syncer and cancel the old updateCollectors goroutine.
-		svc.syncer.Close()
-		svc.syncer = nil
-	}
-
-	svc.cancelSyncBackgroundRoutine()
-
-	// Kick off syncer if we're running it.
-	if intervalMins > 0 && !svc.syncDisabled {
-		syncer, err := svc.syncerConstructor(svc.syncLogger, cfg)
+func (svc *builtIn) syncPreviouslyCaptured() {
+	// Sync existing files in captureDir.
+	var previouslyCaptured []string
+	//nolint
+	_ = filepath.Walk(svc.captureDir, func(path string, info os.FileInfo, err error) error {
 		if err != nil {
-			return errors.Wrap(err, "failed to initialize new syncer")
+			return nil
 		}
-		svc.syncer = syncer
-
-		// Sync existing files in captureDir.
-		var previouslyCaptured []string
-		//nolint
-		_ = filepath.Walk(svc.captureDir, func(path string, info os.FileInfo, err error) error {
-			if err != nil {
-				return nil
-			}
-			if info.IsDir() {
-				return nil
-			}
-			previouslyCaptured = append(previouslyCaptured, path)
+		if info.IsDir() {
 			return nil
-		})
-		svc.syncer.Sync(previouslyCaptured)
-
-		// Validate svc.additionSyncPaths all exist, and create them if not. Then sync files in svc.additionalSyncPaths.
-		svc.syncer.Sync(svc.buildAdditionalSyncPaths())
+		}
+		previouslyCaptured = append(previouslyCaptured, path)
+		return nil
+	})
+	svc.syncer.SyncCaptureFiles(previouslyCaptured)
 
-		// Kick off background routine to periodically sync files.
-		svc.startSyncBackgroundRoutine(intervalMins)
-	}
-	return nil
+	// TODO: Sync existing arbitrary files.
 }
 
 // Sync performs a non-scheduled sync of the data in the capture directory.
-func (svc *builtIn) Sync(_ context.Context, extra map[string]interface{}) error {
+func (svc *builtIn) Sync(_ context.Context, _ map[string]interface{}) error {
```

Comment:
If sync is called (either in this manual Sync() or as part of the scheduled interval) multiple times in succession faster than we can upload existing/queued files, would we spawn multiple goroutines to sync the same files and thus upload them multiple times?